src/heap.rs

```diff
@@ -117,6 +117,14 @@ impl<T: Display> Display for LeftistHeap<T> {
 
 macro_rules! vecdeque {
     ($( $v: expr ),*) => {{
+         let vec = ::std::collections::VecDeque::new();
```

Comment:
This doesn't work since you're trying to use `push_back` on a immutable `VecDeque`, the reason this compiles is because all usages of `vecdeque!` are for empty `VecDeque`s. If you try the following: `vecdeque![0, 1]` it won't compile. So please remove this macro and replace all instances of `vecdeque![]` with `VecDeque::new()`.